### PR TITLE
Acquire yoloxl graph

### PR DESCRIPTION
**Pull request by zichengwang47**
Modified `reshape`, `resize`, `slice` to support acquiring the graph of yolo_xl onnx model on both CPU and GPU.

### CODE REVIEW

File: onnx2torch/node_converters/slice.py [hunk 1]
```diff
@@ -119,7 +126,19 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint:
 @add_converter(operation_type='Slice', version=11)
 @add_converter(operation_type='Slice', version=13)
 def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint: disable=unused-argument
+    constant_params = []
+    params_names = node.input_values[1:]
+    for name in params_names:
+        try:
+            constant_params.append(get_const_value(name, graph).tolist())
+        except KeyError:
+            constant_params.append([])
```
Comment (kinjalpatel27) [inline onnx2torch/node_converters/slice.py]:
```suggestion
        if name in graph.initializers:
            constant_params.append(get_const_value(name, graph).tolist())
        else:
            constant_params.append([])
```

File: onnx2torch/node_converters/slice.py [hunk 2]
```diff
@@ -119,7 +126,19 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint:
...
+    constant_params = constant_params + (4 - len(constant_params)) * [[]]
+
+
     return OperationConverterResult(
-        torch_module=OnnxSlice(),
+        torch_module=OnnxSlice(
+            *constant_params
+        ),
```
Comment (kinjalpatel27) [inline onnx2torch/node_converters/slice.py]:
```suggestion
        torch_module=OnnxSlice(*constant_params),
```

File: onnx2torch/node_converters/slice.py [hunk 3]
```diff
@@ -98,7 +105,7 @@ def _forward():
                 args.append(steps)
 
             return DefaultExportToOnnx.export(_forward, 'Slice', *args, {})
-
+
```
Comment (kinjalpatel27) [inline onnx2torch/node_converters/slice.py]:
```suggestion

```

File: onnx2torch/node_converters/slice.py [hunk 4]
```diff
@@ -119,7 +126,19 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint:
...
 @add_converter(operation_type='Slice', version=13)
 def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint: disable=unused-argument
+    constant_params = []
+    params_names = node.input_values[1:]
+    for name in params_names:
+        try:
+            constant_params.append(get_const_value(name, graph).tolist())
+        except KeyError:
+            constant_params.append([])
+    constant_params = constant_params + (4 - len(constant_params)) * [[]]
```
Comment (kinjalpatel27) [inline onnx2torch/node_converters/slice.py:135]: why is this needed?

File: onnx2torch/node_converters/resize.py [hunk 5]
```diff
@@ -161,8 +168,19 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint:
...
 
     ignore_roi = coordinate_transformation_mode != 'tf_crop_and_resize'
+
+    constant_params = []
+    params_names = node.input_values[1:]
+    for name in params_names:
+        try:
+            constant_params.append(get_const_value(name, graph).tolist())
+        except KeyError:
+            constant_params.append([])
```
Comment (kinjalpatel27) [inline onnx2torch/node_converters/resize.py]:
```suggestion
        if name in graph.initializers:
            constant_params.append(get_const_value(name, graph).tolist())
        else:
            constant_params.append([])
```

File: onnx2torch/node_converters/resize.py [hunk 6]
```diff
@@ -161,8 +168,19 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint:
...
+    constant_params = constant_params + (3 - len(constant_params)) * [[]]
```
Comment (kinjalpatel27) [inline onnx2torch/node_converters/resize.py:179]: why is this needed?

File: onnx2torch/node_converters/reshape.py [hunk 7]
```diff
@@ -41,8 +47,13 @@ def forward(  # pylint: disable=missing-function-docstring
 def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint: disable=unused-argument
     if node.attributes.get('allowzero', 0) == 1:
         raise NotImplementedError('"allowzero=1" is not implemented')
+    
+    try:
+        shape = get_const_value(node.input_values[1], graph).tolist()
+    except:
+        shape = []
```
Comment (kinjalpatel27) [inline onnx2torch/node_converters/reshape.py]:
```suggestion
    if node.input_values[1] in graph.initializers:
        shape = get_const_value(node.input_values[1], graph).tolist()
    else:
        shape = []
```

**Review by kinjalpatel27 — CHANGES_REQUESTED** [review]

File: onnx2torch/node_converters/slice.py (hunk 1 above)
Comment (zichengwang47) [reply onnx2torch/node_converters/slice.py]: I think `get_const_value` already checks if `name` is in `graph.initializers` or in `graph.value_as_node_output`. and in some cases these constants are stored in `graph.value_as_node_output`

File: onnx2torch/node_converters/slice.py (hunk 4 above)
Comment (zichengwang47) [reply onnx2torch/node_converters/slice.py:135]: `onnx.slice` takes in 5 inputs: `data`, `starts`, `ends`, `axes` (optional), `steps` (optional).
For example, if the value of `steps` is not given as the input to a `onnx.slice` node, then the length of `node.input_values[1:]` will just be 3, but `constant_params` needs to have four elements to instantiate the converted torch module.

File: onnx2torch/node_converters/slice.py (hunk 1 above)
Comment (kinjalpatel27) [reply onnx2torch/node_converters/slice.py]: The reason I suggested this `if` condition here is to avoid using the `try, except` block.

**Review by ofershai — APPROVED** [review]

**Review by kinjalpatel27 — APPROVED** [review]

File: onnx2torch/node_converters/slice.py [hunk 8]
```diff
@@ -119,7 +125,17 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint:
 @add_converter(operation_type='Slice', version=11)
 @add_converter(operation_type='Slice', version=13)
 def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint: disable=unused-argument
+    constant_params = []
+    params_names = node.input_values[1:]
+    for name in params_names:
```
Comment (chayim) [inline onnx2torch/node_converters/slice.py:130]: What do you think about using a ilst comprehension as below? This should result in a single statement being translated by python's compiler, resulting in increased efficiency.

```constant_params = [get_const_value(name, graph).tolist() if name in graph.initializers or name in graph._node_output_values else [] for name in params_names]```
